PlagiarismPlugin.inc.php

```diff
@@ -3,26 +3,58 @@
 /**
  * @file PlagiarismPlugin.inc.php
  *
- * Copyright (c) 2003-2021 Simon Fraser University
- * Copyright (c) 2003-2021 John Willinsky
+ * Copyright (c) 2003-2024 Simon Fraser University
+ * Copyright (c) 2003-2024 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
  *
  * @brief Plagiarism plugin
  */
 
 import('lib.pkp.classes.plugins.GenericPlugin');
+import('lib.pkp.classes.db.DAORegistry');
 
 class PlagiarismPlugin extends GenericPlugin {
+
+	/**
+	 * Specify a default integration name for iThenticate service
+	 */
+	public const PLUGIN_INTEGRATION_NAME = 'Plagiarism plugin for OJS/OMP/OPS';
+
+	/**
+	 * Number of seconds EULA details for a context should be cached before refreshing it
+	 */
+	public const EULA_CACHE_LIFETIME = 60 * 60 * 24;
+
+	/**
+	 * Set the value to `true` to enable test mode that will log instead of interacting with 
+	 * iThenticate API service.
+	 */
+	protected const ITHENTICATE_TEST_MODE_ENABLE = false;
+
 	/**
 	 * @copydoc Plugin::register()
 	 */
 	public function register($category, $path, $mainContextId = null) {
 		$success = parent::register($category, $path, $mainContextId);
-		$this->addLocaleData();
 
-		if ($success && $this->getEnabled()) {
-			HookRegistry::register('submissionsubmitstep4form::execute', array($this, 'callback'));
+		$this->addLocaleData();
+		
+		if (!($success || $this->getEnabled())) {
+			return false;	
 		}
+
+		// Need to register both of TemplateManager display and fetch hook as both of these
+		// get called when presenting the submission 
+		HookRegistry::register('TemplateManager::display', [$this, 'addEulaToChecklist']);
+		HookRegistry::register('TemplateManager::fetch', [$this, 'addEulaToChecklist']);
+
+		HookRegistry::register('Submission::add', [$this, 'stampEulaToSubmission']);
+		HookRegistry::register('submissionsubmitstep4form::execute', [$this, 'submitForPlagiarismCheck']);
```

Comment:
The EULA is being added to the checklist (step 1 of submission), but the EULA current version is being stamped on step 4 of submission. Could these possibly be different versions if the user took a few days between step 1 and step 4 of the submission process?

Reply:
> EULA current version is being stamped on step 4 of submission

Not actually , it's being added right after the first step get done and proceed to second step , we have used up a hook `Submission::add` to attach it right after the submission added .